static/app/components/timeline/index.tsx

```diff
@@ -0,0 +1,177 @@
+import {useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import {getFormattedTimestamp} from 'sentry/components/events/interfaces/breadcrumbs/breadcrumb/time/utils';
+import {Tooltip} from 'sentry/components/tooltip';
+import {space} from 'sentry/styles/space';
+import {defined} from 'sentry/utils';
+import type {Color} from 'sentry/utils/theme';
+
+export interface ColorConfig {
+  primary: Color;
+  secondary: Color;
+}
+
+export interface ItemProps {
+  icon: React.ReactNode;
+  timeString: string;
+  title: React.ReactNode;
+  children?: React.ReactNode;
+  colorConfig?: ColorConfig;
+  isActive?: boolean;
+  onClick?: React.MouseEventHandler<HTMLDivElement>;
+  onMouseEnter?: React.MouseEventHandler<HTMLDivElement>;
+  onMouseLeave?: React.MouseEventHandler<HTMLDivElement>;
+  startTimeString?: string;
+}
+
+export function Item({
+  title,
+  children,
+  icon,
+  timeString,
+  startTimeString,
+  colorConfig = {primary: 'gray300', secondary: 'gray200'},
+  isActive = false,
+  onClick,
+  onMouseEnter,
+  onMouseLeave,
+}: ItemProps) {
+  const hasRelativeTime = defined(startTimeString);
+  const placeholderTime = useMemo(() => new Date().toTimeString(), []);
+  const {
+    displayTime,
+    date,
+    timeWithMilliseconds: preciseTime,
+  } = hasRelativeTime
+    ? getFormattedTimestamp(timeString, startTimeString, true)
+    : getFormattedTimestamp(timeString, placeholderTime);
+
+  return (
+    <Row
+      color={colorConfig.secondary}
+      hasLowerBorder={isActive ?? false}
+      onClick={onClick}
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}
+    >
+      <IconWrapper colorConfig={colorConfig} hasIconBorder={isActive ?? false}>
+        {icon}
+      </IconWrapper>
+      <Title color={colorConfig.primary}>{title}</Title>
+      <Timestamp>
+        <Tooltip title={`${preciseTime} - ${date}`}>{displayTime}</Tooltip>
```

Comment:
been trying to skip more wrappers
```suggestion
        <Tooltip title={`${preciseTime} - ${date}`} skipWrapper>{displayTime}</Tooltip>
```